packages/bff/src/services/authorizationService.ts

```diff
@@ -239,6 +239,41 @@ export function authorizationServiceBuilder(
 
       const claims = buildSupportClaims(selfcareId, tenant);
 
+      const { serialized: sessionToken } =
+        await interopTokenGenerator.generateSessionToken(
+          claims,
+          config.supportLandingJwtDuration
+        );
+
+      return sessionToken;
+    },
+    generateJwtFromSaml: async (
+      correlationId: string,
+      samlResponse: string,
+      tenantId: string
+    ): Promise<string> => {
+      validateSamlResponse(samlResponse);
+
+      const { serialized } =
+        await interopTokenGenerator.generateInternalToken();
+
+      const headers = {
+        "X-Correlation-Id": correlationId,
+        Authorization: `Bearer ${serialized}`,
+      };
+
+      const tenant = await tenantProcessClient.tenant.getTenant({
+        params: { id: config.pagoPaTenantId },
+        headers,
```

Comment:
Also here I think there is a difference from what happens in Scala: in the case of the authorization SAML callback the route is unauthenticated, so to call the tenant API it needs to generate an internal token. But in this case, the route is authenticated so you can reuse the token from the request without the need to generate an internal one